sources/api/pluto/src/api.rs

```diff
@@ -1,76 +1,61 @@
-    use snafu::{ResultExt, Snafu};
-
-    #[derive(Debug, Snafu)]
-    pub(crate) enum Error {
-        #[snafu(display("Error calling Bottlerocket API: {}", source))]
-        ApiClient {
-            #[snafu(source(from(apiclient::Error, Box::new)))]
-            source: Box<apiclient::Error>,
-            uri: String,
-        },
-
-        #[snafu(display("Unable to deserialize Bottlerocket settings: {}", source))]
-        SettingsJson { source: serde_json::Error },
-    }
-
-    /// Gets the Bottlerocket settings from the API and deserializes them into a struct.
-    async fn get_settings() -> Result<model::Settings> {
-        let uri = constants::API_SETTINGS_URI;
-        let (_status, response_body) =
-            apiclient::raw_request(constants::API_SOCKET, uri, "GET", None)
-                .await
-                .context(ApiClientSnafu { uri })?;
-
-        serde_json::from_str(&response_body).context(SettingsJsonSnafu)
-    }
-
-    /// Gets the info that we need to know about the EKS cluster from the Bottlerocket API.
-    pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
-        let settings = get_settings().await?;
-        Ok(AwsK8sInfo {
-            region: settings.aws.and_then(|a| a.region).map(|s| s.into()),
-            cluster_name: settings
-                .kubernetes
-                .as_ref()
-                .and_then(|k| k.cluster_name.clone())
-                .map(|s| s.into()),
-            cluster_dns_ip: settings.kubernetes.and_then(|k| k.cluster_dns_ip),
-        })
-    }
+#[derive(Debug, Snafu)]
+pub(crate) enum Error {
+    #[snafu(display("Failed to read configuration file: {}", source))]
+    Read {
+        #[snafu(source(from(std::io::Error, Box::new)))]
+        source: Box<std::io::Error>,
+    },
+    #[snafu(display("Deserialization of configuration file failed: {}", source))]
+    DeserializeToml {
+        #[snafu(source(from(toml::de::Error, Box::new)))]
+        source: Box<toml::de::Error>,
+    },
 }
 
-/// This dummy code is compiled when the `sources` workspace is being compiled for non `aws-k8s-*`
-/// variants.
-#[cfg(not(variant_family = "aws-k8s"))]
-mod inner {
-    use super::*;
-    use snafu::Snafu;
-
-    #[derive(Debug, Snafu)]
-    pub(crate) enum Error {
-        #[snafu(display(
-            "The get_aws_k8s_info function is only compatible with aws-k8s variants"
-        ))]
-        WrongVariant,
-    }
-
-    pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
-        WrongVariantSnafu.fail()
+/// Gets the info that we need to know about the EKS cluster from the configuration file.
+/// Note that we cannot rely on the configuration file being generated as pluto gets called
+/// before template files are fully rendered. (i.e. Pluto self resolves some of the values that
+/// are used in the configuration file). To keep this behavior sane if the configuration file
+/// does not exist, return None and auto populate the region off IMDS
```

Comment:
How do we know that we are getting the right behavior? i.e. how do we know that `cluster_dns_ip` is not actually set in the API if the configuration file is not there?